### ISSUE TITLE
How to preserve gradient after vae?

### ISSUE THREAD

**Issue by zzhang2816** [huggingface/diffusers#2990]
Currently when passing latents to the VAE the latents.requires_grad become false, I wonder is there any method to perseve this graidents for the applications like the Clip guidance?

**Comment by zzhang2816** [comment]
I find there is @torch.no_grad() in the forward function, after I add "with torch.enable_grad():", the problem is solved. Thank you!
